### PR TITLE
External DC authkey disk cache

### PR DESCRIPTION
**Pull request by Qusonann**
- authkeys are cached on disk by default
- cache is invalidated on ban
- added cache warmup

### CODE REVIEW

File: tests/Unit/Tools/BanInvalidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public function test_invalidate(): void
+    {
+        $this->cache
+            ->expects(self::once())
+            ->method(self::METHOD_DEL);
+        $this->invalidator->invalidateIfNeeded(new TGException(TGException::ERR_MSG_USER_BANNED), $this->cache);
+    }
+
+    /**
+     * Проверка что кеш не инвалидируется при прочих ошибках
```
Comment (Ririconj) [inline tests/Unit/Tools/BanInvalidatorTest.php:41]: Language

File: src/Client/InfoObtainingClient/InfoClient.php [hunk 2]
```diff
@@ -88,6 +108,8 @@ public function __construct(BasicClientGeneratorInterface $generator)
     public function login(AuthKey $authKey, ?Proxy $proxy, callable $cb): void
     {
         $this->proxy = $proxy;
+        $this->authKey = $authKey;
+        $this->cache = $this->cacheFactory->generate($authKey);
```
Comment (Ririconj) [inline src/Client/InfoObtainingClient/InfoClient.php:112]: Specify cache type through its name

File: src/Client/InfoObtainingClient/InfoClient.php [hunk 3]
```diff
@@ -537,6 +567,28 @@ private function readPictureFromCurrentDC(
         });
     }
 
+    public function warmup(): void
```
Comment (Ririconj) [inline src/Client/InfoObtainingClient/InfoClient.php:570]: Specify on interface level
All public methods must be declared on interface-level

File: tests/Unit/Tools/BanInvalidatorTest.php (hunk 1 above)
Comment (Qusonann) [reply tests/Unit/Tools/BanInvalidatorTest.php]: Fixed

File: src/Client/InfoObtainingClient/InfoClient.php (hunk 2 above)
Comment (Qusonann) [reply src/Client/InfoObtainingClient/InfoClient.php]: Fixed

File: src/Client/InfoObtainingClient/InfoClient.php (hunk 3 above)
Comment (Qusonann) [reply src/Client/InfoObtainingClient/InfoClient.php:563]: Fixed, added several interfaces

**Pull request merged**